script.rb

```diff
@@ -0,0 +1,93 @@
+module Enumerable
+  def my_each
+    return to_enum(:each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item]) }
+    self
+  end
```

Comment:
#my_each when a block is given when self is a range returns the range itself after calling the 
given block once for each element in self
#my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
#my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self